src/librustdoc/config.rs

```diff
@@ -429,6 +431,14 @@ impl Options {
             }
         });
 
+        let favicon = matches.opt_str("favicon-path").map(|s| PathBuf::from(&s));
+        if let Some(ref favicon) = favicon {
+            if !favicon.is_file() {
+                diag.struct_err("option `--favicon-path` argument must be a file").emit();
```

Comment:
Since this change was suggested for docs.rs, i should note that this won't work over there. The favicon file isn't available to the docs.rs builder when it runs rustdoc (in fact it's not even in the repository), so it can't point to a filename for it to try to load.

What would be ideal is if we could take an arbitrary string here, which would then be pasted in for the favicon path, much like the `#![doc(html_favicon_url)]` attribute we're mimicking. That way docs.rs can pass "`/favicon.ico`" which would link to the correct favicon without having to make sure it can be loaded by the builder.

Reply:
Huuuum... We have to differentiate two kind of paths: URL and locals. What do you want to do in both cases?